src/core/manager-serialize.c

```diff
@@ -186,14 +186,14 @@ int manager_serialize(
                         return r;
         }
 
-        r = fflush_and_check(f);
-        if (r < 0)
-                return log_error_errno(r, "Failed to flush serialization: %m");
-
         r = bus_fdset_add_all(m, fds);
         if (r < 0)
                 return log_error_errno(r, "Failed to add bus sockets to serialization: %m");
 
+        r = finish_serialization_file(f);
```

Comment:
Hmm, I'd still place this in the caller, i.e. the same function where open_serialization_file is called for readability